source/Octopus.Tentacle.Tests.Integration/Util/Builders/Decorators/TentacleServiceDecoratorBuilderExtensions.cs

```diff
@@ -8,26 +9,45 @@ public static class TentacleServiceDecoratorBuilderExtensions
+            recordedUsages = localMethodUsages;
 
-            return builder.RegisterProxyDecorator(latestScriptServiceType, service => MethodUsageProxyDecorator.Create(latestScriptServiceType, service, localTracingStats));
+            return builder.RegisterProxyDecorator(testCase.LatestScriptServiceType, service => MethodUsageProxyDecorator.Create(testCase.LatestScriptServiceType, service, localMethodUsages));
         }
 
-        public static TentacleServiceDecoratorBuilder HookServiceMethod<TService>(this TentacleServiceDecoratorBuilder builder, string methodName, MethodInvocationHook<TService>? preInvocation) where TService : class
-            => HookServiceMethod(builder, methodName, preInvocation, null);
+        public static TentacleServiceDecoratorBuilder HookServiceMethod<TService>(this TentacleServiceDecoratorBuilder builder, string methodName, PreMethodInvocationHook<TService, object> preInvocation) where TService : class
+        //if we aren't hooking the post invocation, we don't care about the response type
+            => HookServiceMethod<TService, object,  object>(builder, methodName, preInvocation, null);
 
-        public static TentacleServiceDecoratorBuilder HookServiceMethod<TService>(this TentacleServiceDecoratorBuilder builder, string methodName, MethodInvocationHook<TService>? preInvocation, MethodInvocationHook<TService>? postInvocation) where TService : class
+        public static TentacleServiceDecoratorBuilder HookServiceMethod<TService, TRequest>(this TentacleServiceDecoratorBuilder builder, string methodName, PreMethodInvocationHook<TService, TRequest> preInvocation) where TService : class
+        //if we aren't hooking the post invocation, we don't care about the response type
+            => HookServiceMethod<TService, TRequest,  object>(builder, methodName, preInvocation, null);
+
+        public static TentacleServiceDecoratorBuilder HookServiceMethod<TService, TRequest, TResponse>(this TentacleServiceDecoratorBuilder builder, string methodName, PreMethodInvocationHook<TService, TRequest>? preInvocation, PostMethodInvocationHook<TService, TResponse>? postInvocation) where TService : class
+            => builder.RegisterProxyDecorator<TService>(service => MethodInvocationHookProxyDecorator<TService, TRequest, TResponse>.Create(service, methodName, preInvocation, postInvocation));
+
+        public static TentacleServiceDecoratorBuilder HookServiceMethod(this TentacleServiceDecoratorBuilder builder, TentacleConfigurationTestCase testCase, string methodName, PreMethodInvocationHook<object, object> preInvocation)
+            => HookServiceMethod(builder, testCase, methodName, preInvocation, null);
+
+        public static TentacleServiceDecoratorBuilder HookServiceMethod(this TentacleServiceDecoratorBuilder builder, TentacleConfigurationTestCase testCase, string methodName, PreMethodInvocationHook<object, object>? preInvocation, PostMethodInvocationHook<object, object>? postInvocation)
```

Comment:
I needed to add support for creating a hooked method based on the latest script service type. This does that via reflection